### PR TITLE
A0-1179: Stateful checks for pallet elections migrations

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
# Description

Goal: we verify whether the storage content has been transferred correctly (using set_temp_storage / get_temp_storage). Additionally, we include these checks in unit tests.

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have added tests
- I have bumped `spec_version` and `transaction_version`

### CODE REVIEW

**Review by krzysztofziobro — APPROVED** [review]
Looks fine

File: pallets/elections/src/migrations/mod.rs [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+    pallet_prelude::StorageVersion, storage::storage_prefix, traits::PalletInfoAccess,
+};
 use frame_support::{traits::OnRuntimeUpgrade, weights::Weight};
+use sp_std::vec::Vec;
 
 pub mod v0_to_v1;
 pub mod v1_to_v2;
 pub mod v2_to_v3;
 
+type ValidatorsVec<T> = Vec<<T as frame_system::Config>::AccountId>;
```
Comment (fbielejec) [inline pallets/elections/src/migrations/mod.rs:14]: could be called just `Validators` imho

**Review by fbielejec — APPROVED** [review]
lgtm

File: pallets/elections/src/migrations/mod.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply pallets/elections/src/migrations/mod.rs]: renamed

**Pull request merged**